src/test/emulators/auth/signUp.spec.ts

```diff
@@ -572,4 +581,273 @@ describeAuthEmulator("accounts:signUp", ({ authApi }) => {
+          expect(decoded!.payload.firebase.identities).to.eql({
+            email: [email],
+          });
+
+          expect(res.body.displayName).to.equal(DISPLAY_NAME);
+          expect(decoded!.payload.name).to.equal(DISPLAY_NAME);
+          expect(decoded!.payload.picture).to.equal(PHOTO_URL);
+          expect(decoded!.payload.email_verified).to.be.true;
+          expect(decoded!.payload).to.have.property("customAttribute").equals("custom");
+          expect(decoded!.payload).to.have.property("sessionAttribute").equals("session");
+        });
+    });
+
+    it("should disable email/password user if set", async () => {
+      await updateConfig(
+        authApi(),
+        PROJECT_ID,
+        {
+          blockingFunctions: {
+            triggers: {
+              beforeCreate: {
+                functionUri: BEFORE_CREATE_URL,
+              },
+            },
+          },
+        },
+        "blockingFunctions"
+      );
+      nock(BLOCKING_FUNCTION_HOST)
+        .post(BEFORE_CREATE_PATH)
+        .reply(200, {
+          userRecord: {
+            updateMask: "disabled",
+            disabled: true,
+          },
+        });
+
+      const email = "me@example.com";
+      const password = "notasecret";
+      await authApi()
+        .post("/identitytoolkit.googleapis.com/v1/accounts:signUp")
+        .send({ email, password })
+        .query({ key: "fake-api-key" })
+        .then((res) => {
+          expectStatusCode(400, res);
+          expect(res.body.error.message).to.equal("USER_DISABLED");
+        });
+    });
+
+    it("should not trigger blocking functions for privileged requests", async () => {
+      await updateConfig(
+        authApi(),
+        PROJECT_ID,
+        {
+          blockingFunctions: {
+            triggers: {
+              beforeCreate: {
+                functionUri: BEFORE_CREATE_URL,
+              },
+              beforeSignIn: {
+                functionUri: BEFORE_SIGN_IN_URL,
+              },
+            },
+          },
+        },
+        "blockingFunctions"
+      );
+      nock(BLOCKING_FUNCTION_HOST)
+        .post(BEFORE_CREATE_PATH)
+        .reply(400)
+        .post(BEFORE_SIGN_IN_PATH)
+        .reply(400);
+
+      const phoneNumber = TEST_PHONE_NUMBER;
+      const displayName = "Alice";
+      await authApi()
+        .post("/identitytoolkit.googleapis.com/v1/accounts:signUp")
+        .set("Authorization", "Bearer owner")
+        .send({ phoneNumber, displayName })
+        .then((res) => {
+          expectStatusCode(200, res);
+
+          // Shouldn't be set for authenticated requests:
+          expect(res.body).not.to.have.property("idToken");
+          expect(res.body).not.to.have.property("refreshToken");
+
+          expect(res.body.displayName).to.equal(displayName);
+          expect(res.body.localId).to.be.a("string").and.not.empty;
+        });
+
+      // Shouldn't trigger nock calls
+      expect(nock.isDone()).to.be.false;
+      nock.cleanAll();
```

Comment:
This should be in a `afterEach` block so that the cleanup still happens even if the test fails somehow